## 全局异常拦截
```php
   public function render($request, Throwable $exception)
    {
        if(Str::lower($request->segment(1)) === 'api'){
            if ($exception instanceof ValidationException) {
                return $this->fail($exception->validator->errors()->first());
            } elseif ($exception instanceof ModelNotFoundException) {
                return $this->fail("一不小心数据走丢了～～～",10003,[],200);
            } else if ($exception instanceof NotFoundHttpException) {
                return $this->fail('路由未找到',10004,[],$exception->getStatusCode());
            } else if ($exception instanceof MethodNotAllowedHttpException) {
                return $this->fail('请求方法不存在',[],10005,$exception->getStatusCode());
            } else if ($exception instanceof UnauthorizedHttpException) { //这个在jwt.auth 中间件中抛出
                return $this->fail('无效的访问令牌',10006,null,401);
            } elseif ($exception instanceof AuthenticationException) { //这个异常在 auth:api 中间件中抛出
                return $this->fail('无效的访问令牌',10006,null,401);
            } elseif ($exception instanceof \Symfony\Component\HttpKernel\Exception\HttpException &&
                $exception->getStatusCode() == 403){
                return $this->fail('没有访问权限，请联系管理员',10007,null,$exception->getStatusCode());
            }
           // return $this->fail($exception->getMessage().' '.$exception->getFile(). ' '.$exception->getLine(),10001,null);
        }
        return parent::render($request, $exception);
//        return parent::render($request, $exception);
    }
```
